src/algorithms/reco/JetReconstruction.cc

```diff
@@ -92,36 +99,23 @@ namespace eicrecon {
 
       m_log->trace("  jet {}: pt = {}, y = {}, phi = {}", i, jets[i].pt(), jets[i].rap(), jets[i].phi());
 
-      // Type = 0 for jets, Type = 1 for constituents
-      // Use PDG values to match jets and constituents
+      // create jet to store in output collection
       edm4eic::MutableReconstructedParticle jet_output = jet_collection->create();
-      jet_output.setType(0);
-      jet_output.setPDG(i);
       jet_output.setMomentum(edm4hep::Vector3f(jets[i].px(), jets[i].py(), jets[i].pz()));
       jet_output.setEnergy(jets[i].e());
       jet_output.setMass(jets[i].m());
 
-      // loop over constituents
+      // link constituents to jet kinematic info
       std::vector<PseudoJet> csts = jets[i].constituents();
       for (unsigned j = 0; j < csts.size(); j++) {
-
-        const double cst_pt = csts[j].pt();
-        m_log->trace("    constituent {}'s pt: {}", j, cst_pt);
-
-        // Type = 0 for jets, Type = 1 for constituents
-        // Use PDG values to match jets and constituents
-        edm4eic::MutableReconstructedParticle cst_output = jet_collection->create();
-        cst_output.setType(1);
-        cst_output.setPDG(i);
-        cst_output.setMomentum(edm4hep::Vector3f(csts[j].px(), csts[j].py(), csts[j].pz()));
-        cst_output.setEnergy(csts[j].e());
-        cst_output.setMass(csts[j].m());
-        //jet_output.addToParticles(cst_output);  // FIXME: global issue with podio reference
+        jet_output.addToParticles(input_collection->at(csts[j].user_index()));
```

Comment:
Maybe an issue for another pull request, but do we want to store all constituents in the particles section? This is ok for for GeneratedJets of course, but right now ReconstructedJets constituents are tracks. And in the future, there will be a mix. Not even sure how to specify this from the factory.

Reply:
I think this is understood. We need to get back to discussion of EDM for the next iteration.

Reply:
There is no mix currently. We use MCParticles converted to ReconstructedParticles as there is no onetoone ralation for the prior.